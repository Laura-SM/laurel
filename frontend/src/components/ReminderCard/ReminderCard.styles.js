import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  cardContainer: {
    width: 350,
    height: 150,
    borderWidth: 2,
    borderColor: '#93B1A7',
    borderRadius: 10,
    justifyContent: 'space-evenly',
    alignItems: 'center',
    marginVertical: 10,
    marginLeft: 20,
    flexDirection: 'row',
  },
  image: {
    width: 100,
    height: 100,
    borderRadius: 10,
    backgroundColor: '#E5E5E5',
  },
  featuresContainer: {
    alignItems: 'flex-start',
    width: 150,
  },
  iconsContainer: {
    width: 170,
    flexDirection: 'row',
    alignItems: 'flex-start',
    justifyContent: 'space-around',
    marginVertical: 10,
  },
});

export default styles;
