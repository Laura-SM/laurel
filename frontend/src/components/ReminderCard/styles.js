import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  cardContainer: {
    width: 350,
    height: 150,
    borderWidth: 2,
    borderColor: '#93B1A7',
    borderRadius: 10,
    justifyContent: 'space-evenly',
    alignItems: 'center',
    marginVertical: 10,
    marginLeft: 20,
    flexDirection: 'row',
  },
  image: {
    width: 120,
    height: 120,
    borderRadius: 10,
    backgroundColor: '#E5E5E5',
  },
  featuresContainer: {
    alignItems: 'flex-start',
    width: 180,
  },
  namesContainer: {
    width: 150,
    flexDirection: 'row',
    justifyContent: 'flex-start',
    marginVertical: 10,
  },
  iconsContainer: {
    width: 180,
    flexDirection: 'row',
    justifyContent: 'flex-start',
    marginVertical: 10,
  },
  iconContainer: {
    margin: 10,
  },
  boldText: {
    fontSize: 18,
    marginHorizontal: 5,
    fontWeight: 'bold',
  },
  text: {
    fontSize: 18,
    marginHorizontal: 5,
  },
});

export default styles;
