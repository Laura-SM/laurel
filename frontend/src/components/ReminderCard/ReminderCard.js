import React from 'react';
import {View, Text, Image} from 'react-native';
import globalStyles from '../../styles/global.styles';
import styles from './ReminderCard.styles';

const ReminderCard = ({plant}) => {
  const currentDate = new Date();
  const currentDateNoTime = new Date(
    Date.UTC(
      currentDate.getFullYear(),
      currentDate.getMonth(),
      currentDate.getDate(),
    ),
  );

  const waterDate = new Date(plant.nextWaterDate).toLocaleDateString();
  const mistDate = new Date(plant.nextMistDate).toLocaleDateString();
  const transplantDate = new Date(
    plant.nextTransplantDate,
  ).toLocaleDateString();

  return (
    <View style={styles.cardContainer}>
      <View style={styles.image} />
      <View style={styles.featuresContainer}>
        <View style={styles.iconsContainer}>
          <Text style={globalStyles.boldText}>{plant.name}</Text>
          <Text style={globalStyles.text}>{plant.room}</Text>
        </View>
        <View style={styles.iconsContainer}>
          {waterDate <= currentDateNoTime.toLocaleDateString() && (
            <Image source={require('../../icons/water24.png')} />
          )}
          {mistDate <= currentDateNoTime.toLocaleDateString() && (
            <Image source={require('../../icons/mist24.png')} />
          )}
          {transplantDate <= currentDateNoTime.toLocaleDateString() && (
            <Image source={require('../../icons/settings24.png')} />
          )}
        </View>
      </View>
    </View>
  );
};

export default ReminderCard;
