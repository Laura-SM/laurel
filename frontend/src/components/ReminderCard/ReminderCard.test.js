import 'react-native';
import React from 'react';
import ReminderCard from './index';
import thunk from 'redux-thunk';
import Store from 'redux-mock-store';
import {Provider} from 'react-redux';
import {render} from '@testing-library/react-native';

describe('Given a ReminderCard component,', () => {
  const mockStore = Store([thunk]);
  const myStore = mockStore({
    plants: [
      {
        name: 'kentia',
        _id: '1a',
        image: 'img.com',
        room: 'bedroom',
        nextWaterDate: '10/06/21',
        nextMistDate: '10/06/21',
        nextTransplantDate: '10/06/21',
      },
      {
        name: 'kentia',
        _id: '1a',
        image: 'img.com',
        room: 'bedroom',
        nextWaterDate: '10/06/21',
        nextMistDate: '10/06/21',
        nextTransplantDate: '10/06/21',
      },
    ],
  });
  const plant = {
    name: 'kentia',
    _id: '1a',
    image: 'img.com',
    room: 'bedroom',
    nextWaterDate: '10/06/21',
    nextMistDate: '10/06/21',
    nextTransplantDate: '10/06/21',
  };

  describe('When ReminderCard function is invoked,', () => {
    test('Then PlantCard component renders correctly', () => {
      const wrapper = render(
        <Provider store={myStore}>
          <ReminderCard plant={plant} />
        </Provider>,
      );
      expect(wrapper).toMatchSnapshot();
    });
  });
});
