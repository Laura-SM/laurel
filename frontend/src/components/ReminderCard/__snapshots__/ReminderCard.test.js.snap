// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a ReminderCard component When invoques PlantCard function Then PlantCard component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#93B1A7",
      "borderRadius": 10,
      "borderWidth": 2,
      "flexDirection": "row",
      "height": 150,
      "justifyContent": "space-evenly",
      "marginLeft": 20,
      "marginVertical": 10,
      "width": 350,
    }
  }
>
  <Image
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Object {
        "backgroundColor": "#E5E5E5",
        "borderRadius": 10,
        "height": 100,
        "width": 100,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "width": 150,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexDirection": "row",
          "justifyContent": "space-around",
          "marginVertical": 10,
          "width": 170,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 18,
            "fontWeight": "bold",
            "marginHorizontal": 20,
          }
        }
      />
      <Text
        style={
          Object {
            "fontSize": 18,
            "marginHorizontal": 20,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexDirection": "row",
          "justifyContent": "space-around",
          "marginVertical": 10,
          "width": 170,
        }
      }
    />
  </View>
</View>
`;

exports[`Given a ReminderCard component, When ReminderCard function is invoked, Then PlantCard component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#93B1A7",
      "borderRadius": 10,
      "borderWidth": 2,
      "flexDirection": "row",
      "height": 150,
      "justifyContent": "space-evenly",
      "marginLeft": 20,
      "marginVertical": 10,
      "width": 350,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "img.com",
      }
    }
    style={
      Object {
        "backgroundColor": "#E5E5E5",
        "borderRadius": 10,
        "height": 100,
        "width": 100,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "width": 150,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexDirection": "row",
          "justifyContent": "space-around",
          "marginVertical": 10,
          "width": 170,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 18,
            "fontWeight": "bold",
            "marginHorizontal": 20,
          }
        }
      >
        kentia
      </Text>
      <Text
        style={
          Object {
            "fontSize": 18,
            "marginHorizontal": 20,
          }
        }
      >
        bedroom
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flexDirection": "row",
          "justifyContent": "space-around",
          "marginVertical": 10,
          "width": 170,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/icons/water24.png",
          }
        }
      />
      <Image
        source={
          Object {
            "testUri": "../../../src/icons/mist24.png",
          }
        }
      />
      <Image
        source={
          Object {
            "testUri": "../../../src/icons/settings24.png",
          }
        }
      />
    </View>
  </View>
</View>
`;
